presto-native-execution/presto_cpp/main/types/HivePrestoToVeloxConnector.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "presto_cpp/main/types/PrestoToVeloxConnector.h"
+#include "presto_cpp/main/types/PrestoToVeloxExpr.h"
+#include "presto_cpp/main/types/TypeParser.h"
+#include "presto_cpp/presto_protocol/ConnectorProtocol.h"
+#include "presto_cpp/presto_protocol/presto_protocol.h"
+#include "velox/connectors/Connector.h"
+#include "velox/connectors/hive/HiveConfig.h"
+#include "velox/connectors/hive/TableHandle.h"
+#include "velox/connectors/hive/iceberg/IcebergDeleteFile.h"
+#include "velox/core/PlanNode.h"
+#include "velox/vector/ComplexVector.h"
+
+namespace facebook::presto {
+
+velox::dwio::common::FileFormat toVeloxFileFormat(
+    const presto::protocol::StorageFormat& format);
+
+velox::connector::hive::HiveColumnHandle::ColumnType toHiveColumnType(
+    protocol::ColumnType type);
+
+std::vector<velox::common::Subfield> toRequiredSubfields(
+    const protocol::List<protocol::Subfield>& subfields);
+
+std::unique_ptr<velox::connector::ConnectorTableHandle> toHiveTableHandle(
+    const protocol::TupleDomain<protocol::Subfield>& domainPredicate,
+    const std::shared_ptr<protocol::RowExpression>& remainingPredicate,
+    bool isPushdownFilterEnabled,
+    const std::string& tableName,
+    const protocol::List<protocol::Column>& dataColumns,
+    const protocol::TableHandle& tableHandle,
+    const protocol::Map<protocol::String, protocol::String>& tableParameters,
+    const VeloxExprConverter& exprConverter,
+    const TypeParser& typeParser);
+
+void validateHiveConfigs(
```

Comment:
Why is this an external method ? It can be made local to the cpp file in an anonymous namespace. Same for all above methods as well. They are only used by HivePrestoToVeloxConnector. Only the public interface of HivePrestoToVeloxConnector should be made public imo.

Reply:
Thanks for the suggestion, moved `validateHiveConfigs` to anonymous namespace and removed it from header. The rest of the functions, `toHiveTableHandle`, `toRequiredSubfields`, `toHiveColumnType` and `toVeloxFileFormat` are also used by the respective functions in `IcebergPrestoToVeloxConnector`, hence they are exposed in the header file.

Reply:
@pramodsatya : It might be better to move IcebergPrestoToVeloxConnector to HivePrestoToVeloxConnector.h(cpp) files as well. wdyt ?

Reply:
@aditi-pandit, thanks for the suggestion, but I feel this would result in a file similar to the current PrestoToVeloxConnector.h(.cpp) files. The intent with this refactor was to breakdown the large file `PrestoToVeloxConnector.cpp`. How do you suggest we proceed with the refactor?

Reply:
Maybe they should be refactored to their own file. PrestoToVeloxConnector being the base and then the individual connectors should be in their own file. So Iceberg would go separately.